pkg/kubelet/config/config.go

```diff
@@ -260,6 +263,14 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 			if ref.Annotations == nil {
 				ref.Annotations = make(map[string]string)
 			}
+
+			if strings.HasPrefix(source, kubetypes.ApiserverSource) {
+				if _, ok := util.Tenant2api[ref.Tenant]; !ok {
+					clientId, _ := strconv.Atoi(source[(len(source) - 1):])
+					util.Tenant2api[ref.Tenant] = clientId
```

Comment:
I think you still need a lock (maybe a RWMutex?) even you have only one writing thread. If I remember correctly Golang runtime will complain if it detects read/write from different goroutines.

Reply:
Also:  1) Not sure how we handle system tenants in kubelet now. Do we need to skip empty string or "system" here?  2) Probably use "toLower" both here and the lookup function to ensure case insensitive?

Reply:
based on this logic I wonder if there's a single-thread place where this map can be filled instead of here to avoid the need of a lock

Reply:
regardless, the delete of map is unavoidable in the future, having the lock in place might be eventually needed anyway so I suggest to have it.

Reply:
> I think you still need a lock (maybe a RWMutex?) even you have only one writing thread. If I remember correctly Golang runtime will complain if it detects read/write from different goroutines.

it seems the lock here is already doing what is needed for this

https://github.com/CentaurusInfra/arktos/blob/0167ab88f8503f1359845a8ef68a88e8a16b755f/pkg/kubelet/config/config.go#L173

Reply:
> Also: 1) Not sure how we handle system tenants in kubelet now. Do we need to skip empty string or "system" here? 2) Probably use "toLower" both here and the lookup function to ensure case insensitive?

1. "system" tenant should still follow the proxy rules. I don't think we allow empty tenant but I'm double checking this.

2. Added the conversion. Although I don't think it has been defined whether tenant name is case sensitive or not.

Reply:
Updates: 

- Add RW lock
- "system" tenant from a certain tenant partition should communicate with the origin tp apiserver
- Tenant by definition should be lower case already. Add toLower to be safe
- Skip empty tenant
- Refactor